runqslower.skel.rs-DELETEME

```diff
@@ -0,0 +1,234 @@
+// SPDX-License-Identifier: (LGPL-2.1 OR BSD-2-Clause)
+//
+// THIS FILE IS AUTOGENERATED BY CARGO-LIBBPF-GEN!
+
+use libbpf_rs::libbpf_sys;
+
+const DATA: &[u8] = include_bytes!("/home/dlxu/dev/libbpf-rs/target/bpf/runqslower.bpf.o");
+
+static mut OBJECT: *mut libbpf_sys::bpf_object = std::ptr::null_mut();
+static mut MAP_START: *mut libbpf_sys::bpf_map = std::ptr::null_mut();
+static mut MAP_MIN_US: *mut libbpf_sys::bpf_map = std::ptr::null_mut();
+static mut MAP_EVENTS: *mut libbpf_sys::bpf_map = std::ptr::null_mut();
+static mut PROG_HANDLE__SCHED_WAKEUP: *mut libbpf_sys::bpf_program = std::ptr::null_mut();
+static mut PROG_HANDLE__SCHED_WAKEUP_LINK: *mut libbpf_sys::bpf_link = std::ptr::null_mut();
+static mut PROG_HANDLE__SCHED_WAKEUP_NEW: *mut libbpf_sys::bpf_program = std::ptr::null_mut();
+static mut PROG_HANDLE__SCHED_WAKEUP_NEW_LINK: *mut libbpf_sys::bpf_link = std::ptr::null_mut();
+static mut PROG_HANDLE__SCHED_SWITCH: *mut libbpf_sys::bpf_program = std::ptr::null_mut();
+static mut PROG_HANDLE__SCHED_SWITCH_LINK: *mut libbpf_sys::bpf_link = std::ptr::null_mut();
```

Comment:
how badly you need to use statics?... That makes creating two skeletons in parallel impossible, no? Why this limitation? I.e., can't the skel config have enough storage for these pointers?

Reply:
> That makes creating two skeletons in parallel impossible, no? 

Do you mean if the user tries to create two `{Open,}FooSkeleton` instances? If yeah, then this would cause issues.

I suppose it's possible to allocate pointers on the heap in `ObjectSkeletonConfig` and then access pointers in .skel.rs via checked array index. Would make things cleaner too.

Reply:
Yes, you can create many independent skeletons with C skeleton, we should probably keep that property here. Not sure you need heap allocation, though. You know at the time of generation how many maps, progs (and links) you need, so code-generated skeleton could just have a fixed-size array of pointers, no? and then yes, just use indices to access them.

Reply:
> could just have a fixed-size array of pointers, no?

I don't think that'll work. All rust structs must be `memmove`-ready -- that's how move semantics are implemented (unless compiler RVOs). It's possible to avoid a heap allocation, but then you'd need to [`Pin`](https://doc.rust-lang.org/stable/std/pin/struct.Pin.html) the `ObjectSkeletonConfig` to prevent instances from being moved. That'd be pretty unergonomic so I think allocating on heap is easier.

Reply:
Oh, ok. Sure, whichever way works. I hoped arrays would be the simplest solution, but it's fine.

Reply:
I guess you could allocate `ObjectSkeletonConfig`s on the heap to avoid the issue. But I still think you need some `Pin` stuff to avoid callers from moving `ObjectSkeletonConfig` out of the box (which could trigger a memmove).